runtime/commonMain/src/kotlinx/serialization/SerialDescriptorBuilder.kt

```diff
@@ -8,7 +8,7 @@ import kotlinx.serialization.internal.*
 
 /**
  * Builder for [SerialDescriptor].
- * The resulting descriptor will be uniquely identified by the given [serialName],
+ * The resulting descriptor will be uniquely identified by the given [serialName] and [typeParameters],
```

Comment:
by the given name, type parameters and elements structure (in a non-recursive manner)

Reply:
Also, it is worth to mention in `SerialDescriptor` documentation to give a hint on how to implement/compare/use SD interface